apps/arena/lib/arena/game_updater.ex

```diff
@@ -557,12 +505,208 @@ defmodule Arena.GameUpdater do
     end)
   end
 
+  ##########################
+  # End Game Initialization
+  ##########################
+
+  ##########################
+  # Game flow. Actions executed in every tick.
+  ##########################
+
+  defp move_players(
+         %{
+           players: players,
+           ticks_to_move: ticks_to_move,
+           external_wall: external_wall,
+           obstacles: obstacles,
+           power_ups: power_ups
+         } = game_state
+       ) do
+    moved_players =
+      players
+      |> Physics.move_entities(
+        ticks_to_move,
+        external_wall,
+        obstacles
+      )
+      |> update_collisions(players, power_ups)
+
+    %{game_state | players: moved_players}
+  end
+
+  # Remove exploded projectiles
+  # Update expired to explode status
+  defp update_projectiles_status(%{projectiles: projectiles} = game_state) do
+    updated_projectiles =
+      projectiles
+      |> Map.reject(fn {_projectile_id, projectile} ->
+        projectile.aditional_info.status == :EXPLODED
+      end)
+      |> Enum.reduce(%{}, fn {projectile_id, projectile}, acc ->
+        if projectile.aditional_info.status == :EXPIRED do
+          acc
+          |> Map.put(
+            projectile_id,
+            put_in(
+              projectile,
+              [:aditional_info, :status],
+              :EXPLODED
+            )
+          )
+        else
+          acc |> Map.put(projectile_id, projectile)
+        end
+      end)
+
+    %{game_state | projectiles: updated_projectiles}
+  end
+
+  defp move_projectiles(
+         %{
+           projectiles: projectiles,
+           players: players,
+           obstacles: obstacles,
+           external_wall: external_wall,
+           ticks_to_move: ticks_to_move
+         } = game_state
+       ) do
+    # We don't want to move recently exploded projectiles
+    {recently_exploded_projectiles, alive_projectiles} =
+      projectiles
+      |> Enum.split_with(fn {_projectile_id, projectile} ->
```

Comment:
You can do `Map.split_with/2` with almost the same result, except the tuple will contain maps so you don't need the `Map.new/1` later on

Reply:
Nice catch! I always forget that Map has some enum functions 😅